File: round_3b_live_coding/deepflow-interview-chatbot/contracts/typescript/src/events.ts
import { z } from "zod";

export const MessageCreatedEventData = z.object({
  conversation_id: z.string(),
  message_id: z.string(),
  role: z.string(),
  message_count: z.number().int(),
});
export type MessageCreatedEventData = z.infer<typeof MessageCreatedEventData>;

export const MessageCreatedEvent = z.object({
  name: z.literal("chat/message.created"),
  data: MessageCreatedEventData,
});
export type MessageCreatedEvent = z.infer<typeof MessageCreatedEvent>;

File: round_3b_live_coding/deepflow-interview-chatbot/contracts/python/chat_contracts/events.py
from pydantic import BaseModel, Field


class MessageCreatedEventData(BaseModel):
    """Payload for the chat/message.created Inngest event."""

    conversation_id: str
    message_id: str
    role: str = Field(
        ..., description="The role of the message author: 'user' or 'assistant'."
    )
    message_count: int = Field(
        ..., description="Total number of messages in the conversation after this one."
    )


class MessageCreatedEvent(BaseModel):
    """Typed envelope for the chat/message.created Inngest event."""

    name: str = Field(default="chat/message.created", frozen=True)
    data: MessageCreatedEventData
